customization-base/src/main/java/com/azure/autorest/customization/ClassCustomization.java

```diff
@@ -285,14 +285,61 @@ public MethodCustomization addMethod(String method) {
         return getMethod(methodSignature);
     }
 
+    /**
+     * Removes a method from this class.
+     *
+     * @param methodNameOrSignature The name or signature of the method being removed.
+     * @return The current ClassCustomization.
+     */
+    public ClassCustomization removeMethod(String methodNameOrSignature) {
```

Comment:
Will this remove all overloads if just a method name is given? It would be good to document the behavior in the javadoc.

Reply:
It will only remove the instance which matches the signature. I've actually been thinking that all methods which take a name or signature should throw if multiple instances are found as it leads to an ambiguous runtime behavior.

I'll update the docs to say this only removes a single instance.